src/platform/qt/library/LibraryController.cpp

```diff
@@ -213,16 +266,46 @@ void LibraryController::loadDirectory(const QString& dir, bool recursive) {
 	mLibraryLoadDirectory(library.get(), dir.toUtf8().constData(), recursive);
 	m_libraryJob.testAndSetOrdered(libraryJob, -1);
 }
+
 void LibraryController::setShowFilename(bool showFilename) {
 	if (showFilename == m_showFilename) {
 		return;
 	}
 	m_showFilename = showFilename;
-	if (m_libraryGrid) {
-		m_libraryGrid->setShowFilename(m_showFilename);
+	m_libraryModel->setShowFilename(m_showFilename);
+	refresh();
+}
+
+void LibraryController::resizeEvent(QResizeEvent*) {
+	resizeTreeView(false);
+}
+
+void LibraryController::resizeTreeView(bool expand) {
+	if (expand) {
+		m_treeView->expandAll();
 	}
-	if (m_libraryTree) {
-		m_libraryTree->setShowFilename(m_showFilename);
+
+	int viewportWidth = m_treeView->viewport()->width();
+	int totalWidth = m_treeView->header()->sectionSizeHint(LibraryModel::COL_CRC32);
```

Comment:
We should probably add a COL_MAX, since there may be more columns later on.

Reply:
Agreed. I had toyed with the idea and then got distracted.